modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java

```diff
@@ -145,6 +173,14 @@
     public ClusterCachesInfo(GridKernalContext ctx) {
         this.ctx = ctx;
 
+        ctx.metric().list(CACHES_MON_LIST, CACHES_MON_LIST_DESC, CacheView.class,
+            l -> cachesMonList = l,
+            l -> cachesMonList = null);
+
+        ctx.metric().list(CACHE_GRPS_MON_LIST, CACHE_GRPS_MON_LIST_DESC, CacheGroupView.class,
+            l -> cachesGrpMonList = l,
+            l -> cachesGrpMonList = null);
+
```

Comment:
List will be inconsistent when someone will disable it in runtime, create new caches and enable again.

Reply:
It'is expected behavior, from my point of view.
It's not easy to build a consistent list if it enabled in runtime.

Transactions list, locks list, can be taken as an example.

Reply:
All lists made consistent after enable/disable.